hypothesis-python/src/hypothesis/strategies/_internal/lazy.py

```diff
@@ -100,6 +103,15 @@ def calc_is_cacheable(self, recur):
     @property
     def wrapped_strategy(self):
         if self.__wrapped_strategy is None:
+            if sideeffect_should_warn():
+                warnings.warn(
```

Comment:
I'd pull this out into a helper function, because we'll also want to call the same logic from `DeferredStrategy`.